pkg/pdfcpu/types/streamdict.go

```diff
@@ -102,6 +103,90 @@ func (sd StreamDict) Image() bool {
 	return true
 }
 
+type DecodeLazyObjectStreamObjectFunc func(c context.Context, s string) (Object, error)
+
+type LazyObjectStreamObject struct {
+	osd         *ObjectStreamDict
+	startOffset int
+	endOffset   int
+
+	decodeFunc    DecodeLazyObjectStreamObjectFunc
+	decodedObject Object
+	decodedError  error
+}
+
+func NewLazyObjectStreamObject(osd *ObjectStreamDict, startOffset, endOffset int, decodeFunc DecodeLazyObjectStreamObjectFunc) Object {
+	return &LazyObjectStreamObject{
+		osd:         osd,
+		startOffset: startOffset,
+		endOffset:   endOffset,
+
+		decodeFunc: decodeFunc,
+	}
+}
+
+func (l *LazyObjectStreamObject) Clone() Object {
+	return &LazyObjectStreamObject{
+		osd:         l.osd,
+		startOffset: l.startOffset,
+		endOffset:   l.endOffset,
+
+		decodeFunc:    l.decodeFunc,
+		decodedObject: l.decodedObject,
+		decodedError:  l.decodedError,
+	}
+}
+
+func (l *LazyObjectStreamObject) PDFString() string {
+	data, err := l.getData()
+	if err != nil {
+		panic(err)
```

Comment:
I was hoping we wouldn't need to resort to panic().
This is really the only instance where pdfcpu would be using panic.

Reply:
I'll change it, so that `PDFString` should not be called internally. However it must be there for the interface to be impemented and the error *could* happen.
Would you rather just log the error and return an empty string?

Reply:
Difficult.. Not in the mood to change the Object interface. OK, then let's panic :/